File: pkg/testutil/pipeline.go
package testutil

import (
	"os"
	"path/filepath"
	"testing"
)

// SetupTestPipelineDir creates a temporary pipeline directory for tests.
// Returns the path to the temporary directory.
// Automatically registers cleanup with t.Cleanup() to remove the directory after the test.
func SetupTestPipelineDir(t *testing.T) string {
	t.Helper()

	tmpDir, err := os.MkdirTemp("", "test-pipelines-*")
	if err != nil {
		t.Fatalf("Failed to create temp pipeline dir: %v", err)
	}

	// Register cleanup to automatically remove directory after test
	t.Cleanup(func() {
		os.RemoveAll(tmpDir)
	})

	// Create packages and test subdirectories for our silly test pipeline
	testDir := filepath.Join(tmpDir, "packages", "test")
	if err := os.MkdirAll(testDir, 0o755); err != nil {
		t.Fatalf("Failed to create packages/test dir: %v", err)
	}

	// Create a silly test pipeline that won't conflict with melange built-ins
	testPipeline := `name: test-hello-package
inputs:
  message:
    description: Message to print
    type: string
    default: "hello world from test pipeline"
pipeline:
  - runs: |
      echo "${{inputs.message}}"
      echo "This is a test pipeline for SecureBuild tests"
`

	if err := os.WriteFile(filepath.Join(testDir, "hello.yaml"), []byte(testPipeline), 0o644); err != nil {
		t.Fatalf("Failed to write test/hello.yaml: %v", err)
	}

	return tmpDir
}

File: securebuild-app/integration/fixtures/pipeline.ts
import * as fs from 'fs';
import * as path from 'path';
import * as os from 'os';

/**
 * Creates a temporary pipeline directory for testing.
 * Returns the path to the temporary directory.
 * Caller should clean up with fs.rmSync(dir, { recursive: true, force: true }).
 */
export function setupTestPipelineDir(): string {
  const tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'test-pipelines-'));

  // Create packages and images subdirectories to match the new structure
  const packagesDir = path.join(tmpDir, 'packages');
  const imagesDir = path.join(tmpDir, 'images');
  fs.mkdirSync(packagesDir, { recursive: true });
  fs.mkdirSync(imagesDir, { recursive: true });

  // Create test subdirectory for our silly test pipeline under packages
  const testDir = path.join(packagesDir, 'test');
  fs.mkdirSync(testDir, { recursive: true });

  // Create a silly test pipeline that won't conflict with melange built-ins
  const testPipeline = `name: test-hello
inputs:
  message:
    description: Message to print
    type: string
    default: "hello world from test pipeline"
pipeline:
  - runs: |
      echo "\${{inputs.message}}"
      echo "This is a test pipeline for SecureBuild tests"
`;

  fs.writeFileSync(path.join(testDir, 'hello.yaml'), testPipeline, 'utf8');

  return tmpDir;
}
